### PR TITLE
added emailing for players and challengers

### PR DESCRIPTION
**Pull request by Devanshusp**
## Overview 
This PR adds emailing when certain actions are performed by the user.

The player receives an email when:
- player signs up
- registers to vote
- sets up election reminder

The challenger receives an email when:
- challenger signs up
- receives a badge from player
- completes challenge

## Test Plan 
rec-screen

## Follow ups 
Still need to create functionality to email challenger when their challenge expires.

### CODE REVIEW

File: src/functions/UserData.js [hunk 1]
```diff
@@ -38,13 +38,30 @@ export async function getUserDatabase() {
...
+}
+
 // call this function when user completes an action
 // ADD email user after certain completed actions
 export async function completedAction(action) {
   const userData = await getUserDatabase();
   const uid = auth.getAuth().currentUser.uid;
   // sign up for election reminders
   if (action === "election reminders") {
+    if (!userData.startedChallenge) {
```
Comment (rickyyx) [inline src/functions/UserData.js]: I assume this is saying that `if it is a player`? 
Should we still send the email to the player who is completing this action, but has also started a challenge (thus a hybrid-user)?

File: src/functions/Email.js [hunk 2]
```diff
@@ -1,17 +1,55 @@
...
+    formData.append("avatar", userData.avatar);
+  }
+
+  // emails player
+  if (emailType === "registered" || emailType === "electionReminder") {
+    const challengerData = await getChallengerDatabase();
+
+    formData.append("firstName", challengerData.name);
+    formData.append("avatar", challengerData.avatar);
+  }
```
Comment (rickyyx) [inline src/functions/Email.js]: nit: If these cases are mutually exclusive, I would separate them into maybe this:
```
if(...) {

} else if (...) {

} else {
// handle wrong input
} 
```

**Review by rickyyx — APPROVED** [review]
Left a few nits - feel free to merge though after resolving the conflict.

File: src/functions/UserData.js (hunk 1 above)
Comment (Devanshusp) [reply src/functions/UserData.js]: Yeah, I talked to Tyler about this; I will be updating this to also send emails to challengers. Tyler had to update the email's content to exclude the reference to a challenger getting a badge specifically when sending this email to a challenger.

**Pull request merged**